app/org/apache/spark/deploy/history/SparkFSFetcher.scala

```diff
@@ -80,6 +91,82 @@ class SparkFSFetcher(fetcherConfData: FetcherConfigurationData) extends Elephant
     logDir
   }
 
+  /**
+   * Returns the namenode address of the  active nameNode
+   * @param conf The Hadoop configuration
+   * @return The namenode address of the active namenode
+   */
+  def getNamenodeAddress(conf: Configuration): String = {
+
+    // check if the fetcherconf has namenode addresses. There can be multiple addresses and
+    // we need to check the active namenode address
+    if(fetcherConfData.getParamMap.get(NAMENODE_ADDRESSES) != null) {
+      val nameNodes: Array[String] = fetcherConfData.getParamMap.get(NAMENODE_ADDRESSES).split(",");
+      for(nameNode <- nameNodes) {
+        if(checkActivation(nameNode)) {
+          return nameNode;
+        }
+      }
+    }
+
+    // if we couldn't find the namenode address in fetcherconf, try to find it in hadoop configuration.
+    var isHAEnabled: Boolean = false;
+    if (conf.get(NAME_SERVICES) != null) {
+      isHAEnabled = true;
+    }
+
+    // check if HA is enabled
+    if (isHAEnabled) {
+      // there can be multiple nameservices separated by ','
+      val nameServices: Array[String] = conf.get(NAME_SERVICES).split(",");
+      for (nameService <- nameServices) {
+        val nameNodeIdentifier: String = conf.get(DFS_HA_NAMENODES + "." + nameService);
+        if (nameNodeIdentifier != null) {
+          // there can be multiple namenode identifiers separated by ','
+          for (nameNodeidentifierValue <- nameNodeIdentifier.split(",")) {
+            val httpValue = conf.get(DFS_NAMENODE_HTTP_ADDRESS + "." + nameService + "." + nameNodeidentifierValue);
```

Comment:
for naming conventions, nameNodeidentifierValue should be nameNodeIdentifierValue